junit/src/main/java/cucumber/runtime/junit/JUnitReporter.java

```diff
@@ -202,7 +203,7 @@ public void fireTestFinished() {
         public void addFailure(Throwable targetException) {
             if (targetException instanceof MultipleFailureException) {
                 addMultipleFailureException((MultipleFailureException) targetException);
-            } else if (isPending(targetException)) {
+            } else if (isAssumptionViolated(targetException)) {
```

Comment:
The Cucumber-JVM statuses `passed`, `skipped`, `undefined`, `pending`, `failed` are mapped to the JUnit statuses `passed`, `skipped`, `failure`, `error`, as:
* `passed` -> `passed`
* `skipped` -> `skipped`
* `failed` -> `failure`
* `undefined`, `pending` -> `skipped` unless in strict mode where they are mapped to `failure`
We do not end up here for a `pending` result unless in strict mode, if in non-strict mode we [go straight to `fireTestIgnored`](https://github.com/cucumber/cucumber-jvm/blob/junit-no-step-notification-option/junit/src/main/java/cucumber/runtime/junit/JUnitReporter.java#L135-L146)

With this PR there is a change in how `AssumptionViolatedException`s are notified, previously they would cause a `fireTestIgnored` in non-strict mode and `fireTestFailure` in strict mode, now they will cause a `fireTestIgnored` in non-strict mode and `fileTestAssumptionFailed` in strict mode. I intended to accept that and view it as a part of the fix to #1007.

Now that I know about the `fileTestAssumptionFailed` I think it should be used instead of `fireTestIgnored` to signal "skipped" for `passed`, `skipped`, `undefined` and `pending` statuses (the latter only in non-strict mode of course), but that is another PR.